presto-parquet/src/main/java/com/facebook/presto/parquet/reader/ParquetReader.java

```diff
@@ -151,6 +154,8 @@ public ParquetReader(
         this.systemMemoryContext = requireNonNull(systemMemoryContext, "systemMemoryContext is null");
         this.currentRowGroupMemoryContext = systemMemoryContext.newAggregatedMemoryContext();
         this.maxReadBlockBytes = requireNonNull(maxReadBlockSize, "maxReadBlockSize is null").toBytes();
+        //We use the value of the maxReadBlockSize to set the max size of the data source buffer
+        this.maxDataSourceBufferSize = this.maxReadBlockBytes;
```

Comment:
@yingsu00 The delta between the internal PR I had for this and this commit is that I am setting the maxDataSourceBufferSize to the maxReadBlockBytes, which is controlled via the `parquet_max_read_block_size` session/config property. IMO this is a good config to reuse, since it closely ties to the approx. amount of data the user wants us to read from the parquet file per fetch.  
Lemme know if your thoughts on this ?